### PR TITLE
Feat/lint using super json

### PR DESCRIPTION
**Pull request by Jakub-Vacek**
## Description

This PR improves DX of Lint command 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have updated the documentation accordingly. For updating Oclif commands documentation use [oclif-dev](https://github.com/oclif/dev-cli#oclif-dev-readme).
- [x] I have read the **CONTRIBUTION_GUIDE** document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/logic/lint.ts [hunk 1]
```diff
@@ -305,6 +154,7 @@ export function formatHuman(
   } else {
     buffer += formatIssues(report.errors);
 
+    console.log('buf', buffer);
```
Comment (lukas-valenta) [inline src/logic/lint.ts:157]: nope

**Review by lukas-valenta — APPROVED** [review]
Looks good

File: src/test/utils.ts [hunk 2]
```diff
@@ -323,10 +324,14 @@ export async function execCLI(
...
       }
-      reject(err.toString());
+      if (rejectOnErr) {
+        reject(err.toString());
+      }
     });
 
-    childProcess.on('error', (err: Error) => reject(err));
+    childProcess.on('error', (err: Error) => {
+      if (rejectOnErr) reject(err);
```
Comment (lukas-valenta) [inline src/test/utils.ts:333]: except this, this not looks good, i missed that

File: src/logic/lint.ts (hunk 1 above)
Comment (Jakub-Vacek) [reply src/logic/lint.ts:157]: //bo 😅

File: src/test/utils.ts (hunk 2 above)
Comment (Jakub-Vacek) [reply src/test/utils.ts:333]: Yep, I missed that too. I was playing with errors in child process and forget to discard my changes.

**Comment by Jakub-Vacek** [comment]
@lukas-valenta Do you want to give it a quick look? I added lint integration tests where I had to fiddle with stdout mocking and exposing it from child process I also added hidden flag to skip map to profile validation - we use lint in a lot of our tests.

File: src/logic/lint.ts [hunk 3]
```diff
@@ -319,20 +320,21 @@ async function prepareLintedMap(
...
+    provider: map.provider,
+    variant: map.variant,
+  });
 
   return {
     ...map,
     ast: mapAst,
-    path,
+    path:
+      map.path ||
```
Comment (lukas-valenta) [inline src/logic/lint.ts:336]:
```suggestion
      map.path ??
```

**Pull request merged**